Eclipse 2018-12: Error running view history - NullPointerException (because of the TableTreeViewer)

## teamexplorer-plugin-2018.12.21-10.06.31.log
### Plugin for Eclipse v.14.134.0.201804261732

2018-12-21 10:06:03,920 ERROR [main] (com.microsoft.tfs.client.eclipse.ui.actions.vc.ViewHistoryAction) Error running view history
java.lang.NullPointerException
	at com.microsoft.tfs.client.common.ui.tasks.vc.ViewHistoryTask.run(ViewHistoryTask.java:61)
	at com.microsoft.tfs.client.eclipse.ui.actions.vc.ViewHistoryAction.doRun(ViewHistoryAction.java:50)
	at com.microsoft.tfs.client.common.ui.framework.action.ExtendedAction.run(ExtendedAction.java:58)
	at org.eclipse.ui.internal.PluginAction.runWithEvent(PluginAction.java:250)
	at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:568)
	at org.eclipse.jface.action.ActionContributionItem.lambda$4(ActionContributionItem.java:400)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4131)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1055)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:3944)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3547)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1173)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1062)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:636)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:563)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:151)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:155)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:400)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:595)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1501)

## Platform log
**Caused by: java.lang.NoClassDefFoundError: org/eclipse/jface/viewers/TableTreeViewer**
[.log](https://github.com/Microsoft/team-explorer-everywhere/files/2702241/default.log)

--- a/source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/vc/BranchHistoryTreeControl.java
+++ b/source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/vc/BranchHistoryTreeControl.java
@@ -20,18 +20,17 @@ import org.eclipse.jface.viewers.ITreeContentProvider;
 import org.eclipse.jface.viewers.LabelProvider;
 import org.eclipse.jface.viewers.StructuredSelection;
 import org.eclipse.jface.viewers.TableLayout;
-import org.eclipse.jface.viewers.TableTreeViewer;
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.jface.viewers.TreeViewerColumn;
 import org.eclipse.jface.viewers.Viewer;
 import org.eclipse.swt.SWT;
-import org.eclipse.swt.custom.TableTree;
 import org.eclipse.swt.graphics.Color;
 import org.eclipse.swt.graphics.Font;
 import org.eclipse.swt.graphics.Image;
 import org.eclipse.swt.layout.FillLayout;
 import org.eclipse.swt.widgets.Composite;
-import org.eclipse.swt.widgets.Table;
-import org.eclipse.swt.widgets.TableColumn;
-
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swt.widgets.TreeColumn;
 import com.microsoft.tfs.client.common.ui.Messages;
 import com.microsoft.tfs.client.common.ui.TFSCommonUIClientPlugin;
 import com.microsoft.tfs.client.common.ui.framework.helper.UIHelpers;
@@ -54,9 +53,7 @@ import com.microsoft.tfs.core.util.Hierarchical;
  */
 public class BranchHistoryTreeControl extends Composite implements IPostSelectionProvider {
 
-    private final TableTreeViewer tableTreeViewer;
-    private final Table table;
-    private final TableTree tableTree;
+    private final TreeViewer tableTreeViewer;
 
     /**
      * SWT table styles that will always be used.
@@ -88,12 +85,11 @@ public class BranchHistoryTreeControl extends Composite implements IPostSelectio
         final FillLayout layout = new FillLayout();
         setLayout(layout);
 
-        tableTreeViewer = new TableTreeViewer(this, TREE_STYLES | (style & OPTIONAL_TREE_STYLES));
-        tableTree = tableTreeViewer.getTableTree();
-        table = tableTree.getTable();
+        tableTreeViewer = new TreeViewer(this, TREE_STYLES | (style & OPTIONAL_TREE_STYLES) | SWT.FULL_SELECTION);
+        final Tree table = tableTreeViewer.getTree();
         table.setHeaderVisible(true);
         table.setLinesVisible(true);
-        createTableColumns(table);
+        createTableColumns();
 
         tableTreeViewer.setContentProvider(new BranchHistoryContentProvider());
         tableTreeViewer.setLabelProvider(new TableLabelProvider());
@@ -103,25 +99,22 @@ public class BranchHistoryTreeControl extends Composite implements IPostSelectio
         tableTreeViewer.setInput(dummyItem);
     }
 
-    private void createTableColumns(final Table table) {
+    private void createTableColumns() {
         final TableLayout tableLayout = new TableLayout();
-        table.setLayout(tableLayout);
+        tableTreeViewer.getTree().setLayout(tableLayout);
 
         tableLayout.addColumnData(new ColumnWeightData(5, true));
-        final TableColumn column1 = new TableColumn(table, SWT.NONE);
+        final TreeColumn column1 = new TreeViewerColumn(tableTreeViewer, SWT.NONE).getColumn();
         column1.setText(Messages.getString("BranchHistoryTreeControl.ColumnHeaderFileName")); //$NON-NLS-1$
-        column1.setResizable(true);
 
         tableLayout.addColumnData(new ColumnWeightData(1, true));
-        final TableColumn column2 = new TableColumn(table, SWT.NONE);
+        final TreeColumn column2 = new TreeViewerColumn(tableTreeViewer, SWT.NONE).getColumn();
         column2.setText(Messages.getString("BranchHistoryTreeControl.ColumnHeaderBranchedFromVersion")); //$NON-NLS-1$
-        column2.setResizable(true);
 
         if (displayLatestVersion) {
             tableLayout.addColumnData(new ColumnWeightData(1, true));
-            final TableColumn column3 = new TableColumn(table, SWT.NONE);
+            final TreeColumn column3 = new TreeViewerColumn(tableTreeViewer, SWT.NONE).getColumn();
             column3.setText(Messages.getString("BranchHistoryTreeControl.ColumnHeaderLatestVersion")); //$NON-NLS-1$
-            column3.setResizable(true);
         }
     }
 

--- a/source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/vc/history/HistoryTreeControl.java
+++ b/source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/vc/history/HistoryTreeControl.java
@@ -25,15 +25,16 @@ import org.eclipse.jface.viewers.IStructuredSelection;
 import org.eclipse.jface.viewers.ITableLabelProvider;
 import org.eclipse.jface.viewers.SelectionChangedEvent;
 import org.eclipse.jface.viewers.TableLayout;
-import org.eclipse.jface.viewers.TableTreeViewer;
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.jface.viewers.TreeViewerColumn;
 import org.eclipse.jface.viewers.Viewer;
 import org.eclipse.swt.SWT;
-import org.eclipse.swt.custom.TableTreeItem;
 import org.eclipse.swt.graphics.Image;
 import org.eclipse.swt.layout.FillLayout;
 import org.eclipse.swt.widgets.Composite;
-import org.eclipse.swt.widgets.Table;
-import org.eclipse.swt.widgets.TableColumn;
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swt.widgets.TreeColumn;
+import org.eclipse.swt.widgets.TreeItem;
 import org.eclipse.ui.IWorkbenchActionConstants;
 import org.eclipse.ui.IWorkbenchPartSite;
 
@@ -69,7 +70,7 @@ public class HistoryTreeControl extends Composite implements ISelectionProvider,
         setLayout(new FillLayout());
 
         tableTreeViewer = new HistoryTableTreeViewer(this, style);
-        final Table table = tableTreeViewer.getTableTree().getTable();
+        final Tree table = tableTreeViewer.getTree();
         AutomationIDHelper.setWidgetID(table, HISTORY_TABLE_ID);
         final TableLayout tableLayout = new TableLayout();
         table.setLayout(tableLayout);
@@ -77,17 +78,17 @@ public class HistoryTreeControl extends Composite implements ISelectionProvider,
         table.setHeaderVisible(true);
 
         tableLayout.addColumnData(new ColumnWeightData(30, 30, true));
-        createTableColumn(table, Messages.getString("HistoryTreeControl.ColumNameChangeset")); //$NON-NLS-1$
+        createTableColumn(Messages.getString("HistoryTreeControl.ColumNameChangeset")); //$NON-NLS-1$
         tableLayout.addColumnData(new ColumnWeightData(40, 40, true));
-        createTableColumn(table, Messages.getString("HistoryTreeControl.ColumnNameChange")); //$NON-NLS-1$
-        tableLayout.addColumnData(new ColumnWeightData(30, 40, true));
-        createTableColumn(table, Messages.getString("HistoryTreeControl.ColumnNameUser")); //$NON-NLS-1$
-        tableLayout.addColumnData(new ColumnWeightData(60, 60, true));
-        createTableColumn(table, Messages.getString("HistoryTreeControl.ColumnNameDate")); //$NON-NLS-1$
+        createTableColumn(Messages.getString("HistoryTreeControl.ColumnNameChange")); //$NON-NLS-1$
+        tableLayout.addColumnData(new ColumnWeightData(50, 50, true));
+        createTableColumn(Messages.getString("HistoryTreeControl.ColumnNameUser")); //$NON-NLS-1$
+        tableLayout.addColumnData(new ColumnWeightData(40, 40, true));
+        createTableColumn(Messages.getString("HistoryTreeControl.ColumnNameDate")); //$NON-NLS-1$
         tableLayout.addColumnData(new ColumnWeightData(100, 100, true));
-        createTableColumn(table, Messages.getString("HistoryTreeControl.ColumnNamePath")); //$NON-NLS-1$
+        createTableColumn(Messages.getString("HistoryTreeControl.ColumnNamePath")); //$NON-NLS-1$
         tableLayout.addColumnData(new ColumnWeightData(100, 100, true));
-        createTableColumn(table, Messages.getString("HistoryTreeControl.ColumnNameComment")); //$NON-NLS-1$
+        createTableColumn(Messages.getString("HistoryTreeControl.ColumnNameComment")); //$NON-NLS-1$
 
         tableTreeViewer.setContentProvider(
             new ContentProvider(UICommandExecutorFactory.newUICommandExecutor(getShell())));
@@ -95,7 +96,7 @@ public class HistoryTreeControl extends Composite implements ISelectionProvider,
 
         contextMenu = new MenuManager("#popup"); //$NON-NLS-1$
         contextMenu.setRemoveAllWhenShown(true);
-        tableTreeViewer.getTableTree().setMenu(contextMenu.createContextMenu(table));
+        table.setMenu(contextMenu.createContextMenu(table));
         contextMenu.addMenuListener(new IMenuListener() {
             @Override
             public void menuAboutToShow(final IMenuManager manager) {
@@ -109,11 +110,10 @@ public class HistoryTreeControl extends Composite implements ISelectionProvider,
                 selection = event.getSelection();
             }
         });
-
     }
 
-    private void createTableColumn(final Table table, final String text) {
-        final TableColumn col = new TableColumn(table, SWT.NONE);
+    private void createTableColumn(final String text) {
+        final TreeColumn col = new TreeViewerColumn(tableTreeViewer, SWT.NONE).getColumn();
         col.setText(text);
         col.pack();
     }
@@ -288,10 +288,10 @@ class ContentProvider extends TreeContentProvider {
     public Object[] getElements(final Object inputElement) {
         if (inputElement instanceof HistoryInput) {
             final HistoryInput historyInput = (HistoryInput) inputElement;
-            final Iterator iter = historyInput.queryHistory();
-            final ArrayList changesets = new ArrayList();
+            final Iterator<Changeset> iter = historyInput.queryHistory();
+            final ArrayList<Changeset> changesets = new ArrayList<Changeset>();
             while (iter.hasNext()) {
-                HistoryManager.addChild(changesets, (Changeset) iter.next(), historyInput.isSlotMode());
+                HistoryManager.addChild(changesets, iter.next(), historyInput.isSlotMode());
             }
             return changesets.toArray();
         }
@@ -419,23 +419,21 @@ class LabelProvider implements ITableLabelProvider {
         }
         return padding.toString();
     }
-
 }
 
-class HistoryTableTreeViewer extends TableTreeViewer {
+class HistoryTableTreeViewer extends TreeViewer {
 
     public HistoryTableTreeViewer(final Composite parent, final int style) {
-        super(parent, style);
+        super(parent, style | SWT.FULL_SELECTION);
     }
 
     public int getLevel(final Changeset c) {
         int level = 1;
-        TableTreeItem item = (TableTreeItem) findItem(c);
+        TreeItem item = (TreeItem) findItem(c);
         while (item.getParentItem() != null) {
             item = item.getParentItem();
             level++;
         }
         return level;
     }
-
 }

--- a/source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/vc/properties/branches/BranchesPropertiesControl.java
+++ b/source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/vc/properties/branches/BranchesPropertiesControl.java
@@ -6,17 +6,16 @@ package com.microsoft.tfs.client.common.ui.controls.vc.properties.branches;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.jface.viewers.ColumnWeightData;
 import org.eclipse.jface.viewers.TableLayout;
-import org.eclipse.jface.viewers.TableTreeViewer;
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.jface.viewers.TreeViewerColumn;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.custom.StackLayout;
-import org.eclipse.swt.custom.TableTree;
 import org.eclipse.swt.layout.GridData;
 import org.eclipse.swt.layout.GridLayout;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Label;
-import org.eclipse.swt.widgets.Table;
-import org.eclipse.swt.widgets.TableColumn;
-
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swt.widgets.TreeColumn;
 import com.microsoft.tfs.client.common.ui.Messages;
 import com.microsoft.tfs.client.common.ui.controls.generic.BaseControl;
 import com.microsoft.tfs.core.clients.versioncontrol.BranchHistory;
@@ -24,10 +23,7 @@ import com.microsoft.tfs.core.clients.versioncontrol.BranchHistoryTreeItem;
 import com.microsoft.tfs.core.clients.versioncontrol.soapextensions.Item;
 
 public class BranchesPropertiesControl extends BaseControl {
-    private final Table table;
-    private final TableTree tableTree;
-
-    private final TableTreeViewer tableTreeViewer;
+    private final TreeViewer tableTreeViewer;
     private final StackLayout stackLayout;
     private final Composite branchHistoryComposite;
     private final Composite noBranchHistoryComposite;
@@ -102,14 +98,13 @@ public class BranchesPropertiesControl extends BaseControl {
         branchesLabel.setLayoutData(new GridData(GridData.FILL, GridData.CENTER, true, false));
         branchesLabel.setText(Messages.getString("BranchesPropertiesControl.BranchesLabelText")); //$NON-NLS-1$
 
-        tableTreeViewer = new TableTreeViewer(branchHistoryComposite, SWT.BORDER);
-        tableTree = tableTreeViewer.getTableTree();
-        table = tableTree.getTable();
+        tableTreeViewer = new TreeViewer(branchHistoryComposite, SWT.BORDER | SWT.FULL_SELECTION);
+        final Tree table = tableTreeViewer.getTree();
         table.setHeaderVisible(true);
         table.setLinesVisible(true);
-        createTableColumns(table);
+        createTableColumns();
 
-        tableTree.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, true));
+        table.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, true));
 
         tableTreeViewer.setContentProvider(new BranchesPropertiesContentProvider());
         tableTreeViewer.setLabelProvider(new BranchesPropertiesLabelProvider());
@@ -147,25 +142,20 @@ public class BranchesPropertiesControl extends BaseControl {
         this.layout();
     }
 
-    private void createTableColumns(final Table table) {
+    private void createTableColumns() {
         final TableLayout tableLayout = new TableLayout();
-        table.setLayout(tableLayout);
+        tableTreeViewer.getTree().setLayout(tableLayout);
 
         tableLayout.addColumnData(new ColumnWeightData(5, true));
-        final TableColumn column1 = new TableColumn(table, SWT.NONE);
+        final TreeColumn column1 = new TreeViewerColumn(tableTreeViewer, SWT.NONE).getColumn();
         column1.setText(Messages.getString("BranchesPropertiesControl.ColumNameFileName")); //$NON-NLS-1$
-        column1.setResizable(true);
-
-        tableLayout.addColumnData(new ColumnWeightData(3, true));
-        final TableColumn column2 = new TableColumn(table, SWT.NONE);
-        column2.setText(Messages.getString("BranchesPropertiesControl.ColumnNameChange")); //$NON-NLS-1$
-        column2.setResizable(true);
 
         tableLayout.addColumnData(new ColumnWeightData(1, true));
-        final TableColumn column3 = new TableColumn(table, SWT.NONE);
+        final TreeColumn column2 = new TreeViewerColumn(tableTreeViewer, SWT.NONE).getColumn();
+        column2.setText(Messages.getString("BranchesPropertiesControl.ColumnNameChange")); //$NON-NLS-1$
+
+        tableLayout.addColumnData(new ColumnWeightData(1, true));
+        final TreeColumn column3 = new TreeViewerColumn(tableTreeViewer, SWT.NONE).getColumn();
         column3.setText(Messages.getString("BranchesPropertiesControl.ColumnNameBranchedFrom")); //$NON-NLS-1$
-        column3.setResizable(true);
-
     }
-
 }

